src/mongo/util/net/ssl_manager.cpp

```diff
@@ -831,39 +832,31 @@ namespace mongo {
     }
 
     SSLConnection* SSLManager::connect(Socket* socket) {
-        SSLConnection* sslConn = new SSLConnection(_clientContext, socket, NULL, 0);
-        ScopeGuard sslGuard = MakeGuard(::SSL_free, sslConn->ssl);
-        ScopeGuard bioGuard = MakeGuard(::BIO_free, sslConn->networkBIO);
+        std::unique_ptr<SSLConnection> sslConn = stdx::make_unique<SSLConnection>(_clientContext, socket, (const char*)NULL, 0);
```

Comment:
Understood. The consensus from the CR is that there needs to be another version for the backport. I've submitted another PR on the 3.0 branch which uses auto_ptr instead. https://github.com/mongodb/mongo/pull/950